pkg/landscape/app/group-view.hoon

```diff
@@ -135,7 +146,28 @@
     :_  state
     (fact:io group-view-update+!>(`update:view`[%initial joining]) /all ~)^~
   :-  (fact:io group-view-update+!>([%hide rid]) /all ~)^~
-  state(joining (~(put by joining) rid request(hidden %.y)))
+  state(joining (~(put by joining) rid request))
+::
+++  is-tracking
+  |=  her=ship
+  ^-  ?
+  %+  lien  ~(tap in ~(key by joining))
+  |=([him=ship name=term] =(her him))
+::
+++  breach
```

Comment:
Feels kind of strange to me to have all this logic in group-view. The *-view agents have always held the role in my mind of thin in/output aggregation layer tailored towards frontends. Breach handling, request tracking etc, don't feel like they fit that very well, might be more at home in the pull-hook. But then, maybe all these things are too entangled for that to be feasible?

Reply:
> more at home in the pull-hook
Right, but the motivating use case for tracking breaches is because poke handling is funny across a breach, and if you send a poke to an old life because you aren't up-to-date on chain state yet, that poke will never get (n)acked, permanently halting the group join flow. More general breach tracking can wait until there's better tooling for this (+on-rift).
> thin IO aggregation layer
This actually isn't how it works in practice because any long term UI state that is per-ship, must live in -view agents, under the store/hook/view agent model. in my mind, the hooks are the most principled part of the trichotomy and should serve as shims until subscription reform becomes real. When/if subscription reform/shrub/??? lands I'd like to seriously look at making the landscape codebase more monolithic